features/map.go

```diff
@@ -129,37 +147,61 @@ func haveMapType(mt ebpf.MapType) error {
 		return err
 	}
 
+	err = checkMapCreation(mt, createMapTypeAttr(mt))
+
+	mc.mapTypes[mt] = err
+
+	return err
+}
+
+func haveMapFlags(mt ebpf.MapType, flags uint32) error {
+	cacheEntry := mapFlagCacheEntry{mt, flags}
+
+	mc.Lock()
+	defer mc.Unlock()
+	err, ok := mc.mapFlags[cacheEntry]
+	if ok {
+		return err
+	}
+
+	attr := createMapTypeAttr(mt)
+	attr.MapFlags |= flags
+	err = checkMapCreation(mt, attr)
+
+	mc.mapFlags[cacheEntry] = err
+
+	return err
+}
+
+func checkMapCreation(mt ebpf.MapType, attr *sys.MapCreateAttr) error {
 	fd, err := sys.MapCreate(createMapTypeAttr(mt))
```

Comment:
```suggestion
	fd, err := sys.MapCreate(attr)
```
Flags won't get set otherwise.

Reply:
Fixed in last commit